components/feature/toolbar/src/test/java/mozilla/components/feature/toolbar/ToolbarPresenterTest.kt

```diff
@@ -101,6 +106,110 @@ class ToolbarPresenterTest {
         verify(toolbar).siteSecure = Toolbar.SiteSecurity.INSECURE
     }
 
+    @Test
+    fun `render overridden web extension action from tabs`() {
+        val mockedBrowserButton = mock<WebExtensionToolbarAction>()
+        val mockedBrowserAction = mock<WebExtensionBrowserAction>()
+        val toolbar: Toolbar = mock()
+        val extensions: Map<String, WebExtensionState> = mapOf(
+            "id" to WebExtensionState("id", "url", mockedBrowserAction)
+        )
+        val store = spy(
+            BrowserStore(
```

Comment:
If we want to test that this overrides the BrowserAction for the Tab, we should also add a global one to the store so we can verify we added the right one (see my comment above on the impl.)

Reply:
Test added!